### PR TITLE
refactor: explicitly support partial settings and compare changes in full settings (as opposed to comparing settings to updates)

### PR DESCRIPTION
**Pull request by hkang1**
## Description

It's not clear if the current `updateSettings` is intended to support partial settings, so explicitly making it so. Seen some PRs using it as partial updates.

UPDATE: There was a typing that allowed partial settings updates but the code was not intentionally supporting it. Now explicitly supporting partial settings updates and correcting the types throughout the codebase.

## Test Plan

- [ ] general sanity check across the UI where `useSettings` is used. Generally pages with tables and filters.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: webui/react/src/components/Table/InteractiveTable.tsx [hunk 1]
```diff
@@ -443,7 +446,7 @@ const InteractiveTable = <T extends object>({
 
       if (isEqual(newSettings, settings)) return;
 
-      updateSettings(newSettings);
+      updateSettings(newSettings as Partial<S>);
```
Comment (hkang1) [inline webui/react/src/components/Table/InteractiveTable.tsx:449]: not sure why but without this, it reports:
```
TS2345: Argument of type 'Partial ' is not assignable to parameter of type 'Partial'.
```

Comment (ClaireNeveu) [reply webui/react/src/components/Table/InteractiveTable.tsx:449]: `Partial<InteractiveTableSettings>` !== `Partial<S>`. If the caller uses a different value for `S` this code is incorrect.

Comment (ClaireNeveu) [reply webui/react/src/components/Table/InteractiveTable.tsx:449]: Are you sure you meant to make this component generic over the settings type?

Comment (hkang1) [reply webui/react/src/components/Table/InteractiveTable.tsx:449]: > `Partial<InteractiveTableSettings>` !== `Partial<S>`. If the caller uses a different value for `S` this code is incorrect.

Even if `S` is defined as `S extends InteractiveTableSettings`? Doesn't that ensure that whatever `S` type you provide needs to conform to `InteractiveTableSettings` at the minimum?

Comment (hkang1) [reply webui/react/src/components/Table/InteractiveTable.tsx:449]: > Are you sure you meant to make this component generic over the settings type?

uh... no? 😆 

lemme take a crack again with just using the `InteractiveTableSettings` type.

Comment (ClaireNeveu) [reply webui/react/src/components/Table/InteractiveTable.tsx:449]: `S` can have more fields than `InteractiveTableSettings` which the function could rely on which is why this is not valid as written (I know this is a special case due to `Partial` but typescript doesn't account for that).

**Pull request merged**